src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.StatementResult.cs

```diff
@@ -32,11 +32,9 @@ public override bool ContainingScopeHasAsyncKeyword()
 
                 return node switch
                 {
-                    AccessorDeclarationSyntax _ => false,
                     MethodDeclarationSyntax method => method.Modifiers.Any(SyntaxKind.AsyncKeyword),
-                    ParenthesizedLambdaExpressionSyntax lambda => lambda.AsyncKeyword.Kind() == SyntaxKind.AsyncKeyword,
-                    SimpleLambdaExpressionSyntax lambda => lambda.AsyncKeyword.Kind() == SyntaxKind.AsyncKeyword,
-                    AnonymousMethodExpressionSyntax anonymous => anonymous.AsyncKeyword.Kind() == SyntaxKind.AsyncKeyword,
+                    LocalFunctionStatementSyntax localFunction => localFunction.Modifiers.Any(SyntaxKind.AsyncKeyword),
```

Comment:
added this for completeness.  figuring out how to add a test for this.

Reply:
Just making sure that the lambda syntaxes were intentionally changed to always return false.

Reply:
lambdas are subclasses of AnonymousFunctionExpressionSyntax .